core/src/main/java/com/linecorp/armeria/internal/RequestContextUtil.java

```diff
@@ -49,5 +49,18 @@ public static IllegalStateException newIllegalContextPushingException(
                 "unexpected thread or forgetting to close previous context.");
     }
 
+    /**
+     * Removes the {@link RequestContext} in the thread-local if exists and returns {@link SafeCloseable} which
+     * pushes it back to the thread-local.
```

Comment:
How about mentioning the specific use case of this method? We could then remove the duplicate comments in this pull request.

Reply:
Ah that's a good idea. Let me add it.